versiondb/memiavl/snapshot.go

```diff
@@ -0,0 +1,265 @@
+package memiavl
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ledgerwatch/erigon-lib/mmap"
+)
+
+func WriteSnapshot(root Node, version int64, nodesFile, keysFile, valuesFile, metadataFile string) error {
+	fpNodes, err := os.OpenFile(nodesFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
```

Comment:
make `os.O_WRONLY|os.O_CREATE|os.O_TRUNC` and `0o600` as const?

Reply:
added a `createFile` utility to reduce repetition.